pdns/pdns_recursor.cc

```diff
@@ -2846,7 +2846,7 @@ static void handleNewUDPQuestion(int fd, FDMultiplexer::funcparam_t& var)
           }
 
           if(g_weDistributeQueries) {
-            distributeAsyncFunction(data, boost::bind(doProcessUDPQuestion, data, fromaddr, dest, source, destination, tv, fd, proxyProtocolValues));
+            distributeAsyncFunction(data, [&]{ return doProcessUDPQuestion(data, fromaddr, dest, source, destination, tv, fd, proxyProtocolValues); });
```

Comment:
This version is capturing a `thread_local` data by reference and passing it to a function that will get executed by a different thread, and also passing `proxyProtocolValues` by reference to that same thread while the initial thread might be altering it at the same time. 
```suggestion
            distributeAsyncFunction(data, [data = data, fromaddr, dest, source, destination, tv, fd, proxyProtocolValues]() mutable { return doProcessUDPQuestion(data, fromaddr, dest, source, destination, tv, fd, proxyProtocolValues); });
```

Perhaps we should stop using a `thread_local` data here, as that optimization still results in a copy when we are distributing anyway.